src/charm_state.py

```diff
@@ -85,6 +167,24 @@ def webserver_config(self) -> WebserverConfig:
             else None,
         )
 
+    @property
+    def flask_config(self) -> dict[str, str | int | bool]:
+        """Get the value of the flask_config charm configuration.
+
+        Returns:
+            The value of the flask_config charm configuration.
+        """
+        return self._flask_config.copy()
```

Comment:
A minor question, is this to make sure that mutation doesn't happen?

Reply:
Yes, it's, since there's not frozen dict in Python.

Reply:
Is a shallow copy enough here?

Reply:
Yes, the `flask_config` is always a mapping from string to string, integer or boolean.